GitCommands/packages.config

```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="SmartFormat.NET" version="2.0.0.0" targetFramework="net461" />
+  <package id="StyleCop.Analyzers" version="1.1.0-beta006" targetFramework="net461" developmentDependency="true" />
```

Comment:
@sharwell are there significant difference between 1.0.2 (official) and 1.1.0-beta?

Reply:
Yes, 1.0.2 only supports C# 6, while 1.1 supports C# 7.x (with 1.1.0-beta006 supporting through 7.2).

Reply:
Further, the package doesn't provide any binaries or other files in the referencing project's output folder. It only applies at design and build time (much like the `JetBrains.Annotations` package).

Reply:
Yeah, we've been using old StyleCop, now switching it to the Analyser.
Looks pretty cool, though Nuget complains about referenced beta packages (we create a Nuget package which references those).
@sharwell any ETA for 1.1.0 release package?

Reply:
@RussKie you may like to vote/sub here https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2585

Reply:
> Looks pretty cool, though Nuget complains about referenced beta packages (we create a Nuget package which references those).

This is a sign that you likely have an incorrect reference to StyleCop Analyzers in your package. If you use packages.config for NuGet, this warning should not appear. If you use PackageReference, you need to make sure to set the PrivateAssets attribute like this, and the warning will no longer appear:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/c7254023938e4e6cc5bb03caf6da0ba814845203/StyleCop.Analyzers/Directory.Build.props#L49

Reply:
Perhaps I wasn't clear enough.

Here's my nuspec (trimmed down and simplified for demo purposes):

```
<?xml version="1.0"?>
<package>
  <metadata>
    <id>My.Package</id>
    <version>$version$</version>
    <dependencies>
      <dependency id="StyleCop.Analyzers" version="[1.1.0-beta006,)" />
    </dependencies>
  </metadata>
  <files>
    <file src="..\My.ruleset" target="content" />
    <file src="Install.ps1" target="tools" />
  </files>
</package>
```

When I package it I get the following warnings:

```
WARNING: Issue: Prerelease dependency in stable package.
WARNING: Description: A stable release of a package should not have a prerelease dependency.
WARNING: Solution: Either modify the version spec of dependency "StyleCop.Analyzers [1.1.0-beta006, )" or update the version field in the nuspec.
```

Reply:
@RussKie It looks like this NuGet package is used for in-house development purposes only, and never becomes a downstream dependency. The `-dev` suffix is the same as any other prerelease suffix (e.g. `-alpha` or `-beta`), and using it will prevent NuGet from warning about prerelease dependencies. The `developmentDependency` line will prevent the package from being considered in downstream packaging, which prevents the `-dev` prerelease suffix from triggering warnings there.

As long as you aren't planning to "release" a "stable" version of **My.Package**, there's essentially no downside to this approach and you can leave it in a perpetual state of development.